extensions/eclipse-che-theia-user-preferences/src/node/che-theia-preferences-synchronizer.ts

```diff
@@ -73,17 +93,19 @@ export class CheTheiaUserPreferencesSynchronizer {
      * Updates Theia user preferences which stored in Che
      */
     protected async updateTheiaUserPreferencesInCheSettings(): Promise<void> {
-        let userPreferences = await readFile(THEIA_USER_PREFERENCES_PATH, 'utf8');
+        let userPreferencesContent = await readFile(THEIA_USER_PREFERENCES_PATH, 'utf8');
         try {
             // check json validity and remove indents
-            userPreferences = JSON.stringify(JSON.parse(userPreferences));
+            const userPreferences = JSON.parse(userPreferencesContent);
+            this.fireUserPreferencesModify(userPreferences);
+            userPreferencesContent = JSON.stringify(userPreferences);
```

Comment:
Sorry, why do you override `userPreferencesContent = JSON.stringify(userPreferences);` ?  You have already got  content like string in the code above:
`let userPreferencesContent = await readFile(THEIA_USER_PREFERENCES_PATH, 'utf8');`

Reply:
The origin logic was not modified in this piece of code, just extracted parsed JSON object to fire it in event. 1) we read `settings.json`; 2) validate it through `JSON.parse`; 3) fire event with parsed preferences (was added); 4) call `JSON.stringify`, I suppose just for formatting JSON string; 5) update user preferences with new value.

Agree, looks a bit odd, but decided not to remove this line.